csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -240,6 +288,14 @@ void AllocationOrderInferencer::handle(TernaryOp* op) {
 }
 
 void AllocationOrderInferencer::handle(PadOp* op) {
+  auto* out = dynamic_cast<TensorView*>(op->out());
+  auto* in = dynamic_cast<TensorView*>(op->in());
+  // Note: `out` from pad has rfactor domain that cannot be mapped back to
+  // `in`'s domain. Hence we use `out`'s root domain to match permutation.
+  propagateAllocationOrder(in, out, out->getRootDomain());
```

Comment:
I don't understand why this is necessary. 

> `out` from pad has rfactor domain that cannot be mapped back to `in`'s domain

This is true, but why does it matter? The other usage of `propagateAllocationOrder` passes the rfactor domain of the out tensor. Why does pad need to pass the root domain?

Reply:
usually we have:
producer_rfactor_domain: {i0, i1, i2}
with allocation order, we'll construct a producer alloc domain as `permutation({i0, i1, i2})`

with a pointwise, we'll have consumer_rfactor_domain: {i0, i1, i2}, so it can be mapped to the above permutation to get consumer alloc domain

with pad, what we have here is that:
consumer_root_domain: {i0, i1, i2}
consumer_rfactor_domain: {(i0+-pad), (i1+-pad), (i2+-pad)}

We can no longer map consumer_rfactor_domain to its producer alloc domain, hence we need to special case on pad.